scene/resources/2d/SCsub

```diff
@@ -3,5 +3,20 @@ from misc.utility.scons_hints import *
 
 Import("env")
 
-env.add_source_files(env.scene_sources, "*.cpp")
-env.add_source_files(env.scene_sources, "skeleton/*.cpp")
+env.add_source_files(env.scene_sources, "navigation_mesh_source_geometry_data_2d.cpp")
+env.add_source_files(env.scene_sources, "navigation_polygon.cpp")
+env.add_source_files(env.scene_sources, "polygon_path_finder.cpp")
+env.add_source_files(env.scene_sources, "tile_set.cpp")
+
+if not env["disable_physics_2d"]:
+    env.add_source_files(env.scene_sources, "capsule_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "circle_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "concave_polygon_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "convex_polygon_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "rectangle_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "segment_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "separation_ray_shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "shape_2d.cpp")
+    env.add_source_files(env.scene_sources, "world_boundary_shape_2d.cpp")
```

Comment:
Wouldn't it make sense to move them to a `physics` folder?

Reply:
Wouldn't the diff noise in includes be a problem? I remember you talking about this in the advanced GUI PR.

Reply:
It would, I'm not 100% sure what's the best tradeoff.

My issue with the advanced GUI PR isn't so much the diff noise but the `advanced` name, which feels quite arbitrary and doesn't make it easy to know where a given Control would be. While here if the folder is `physics` or `shapes`, once contributors understand that 2D shapes are there, it's easy to remember IMO.

But let's leave it like this for now and see if we want to refactor later.